share/migrations/0031_populate_rawdata_app_labels.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.7 on 2016-08-26 15:12
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def set_app_labels(apps, schema_editor):
+    RawData = apps.get_model('share', 'RawData')
```

Comment:
This, or something like it, will be **much** faster
```SQL
UPDATE
  rawdata
SET 
  rawdata.app_label = REPLACE(shareuser.robot, 'providers.', '')
JOIN
  shareuser
ON
  rawdata.source = shareuser.id
```

pinging @cwisecarver for second thoughts